docker/docker-compose.yml

```diff
@@ -0,0 +1,56 @@
+version: '2'
+services:
+
+  # The leastauthority.com website, including handling of the signup forms.
+  web:
+    # See build.sh
+    image: 'leastauthority.com/web'
+
+    ports:
+      # The web server in the container listens for HTTPS on 8443.
+      # Publish that to the world on the standard HTTPS port.
+      - '443:8443'
+
+    volumes:
+
+      # The web server generates a lot of random output right now.
+      # Subscription logs, server logs, etc.  Give it a safe place to
+      # dump all of that data.
+      - 'web-data:/app/data'
+
+      # This is unfortunate.  It's here just so we can get the signup
+      # furl generated at flappserver startup time.  The web server
+      # reads the signup.furl file that the flapp server writes into
+      # this volume.
+      - 'flapp-data:/app/flapp-data'
+
+    depends_on:
+
+      # Start the flapp server first so the signup.furl is written.
+      # This is subject to startup race-conditions since Docker has no
+      # idea when signup.furl is written: it only knows that it has
+      # _started_ the flapp container.  Experiments suggest the flapp
+      # server typically wins.  Ideally some alternate mechanism could
+      # be used to convey the signup furl between processes (in other
+      # words, fix the application to not use the filesystem for RPC).
+        - "flapp"
```

Comment:
The systemd `READY=1` example is roughly along the lines of my preferred solution.  Generalizing, the application should emit readiness notification in a way that dependencies can consume it.  Using systemd as a bus for this information seems like it makes sense if systemd is already managing the units.  It presents a uniform interface for delivering the notification and a system for reacting to it (ie, dependent units can start after readiness is signaled).

I'm not sure if there's a way to apply this to a group of Docker-based containers.  But perhaps there's a similar bus available in one of the container management tools.  If not, we could create one ourselves (or we could deliver the notification directly to the web server with no bus, if we don't think we're going to have other use-cases for this feature - and I don't know of any right now).

Stepping back, though, a broader question that I want to discuss is whether the flappserver will continue to be a good solution or if something simpler can get the job done.  I don't know if I'm aware of all of the design considerations that went into choosing the current approach so it'd be great to discuss them before making any major changes.